src/System.IO.Pipelines/Pipe.Awaitable.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.IO.Pipelines
+{
+    internal partial class Pipe
+    {
+        private struct Awaitable
```

Comment:
Nit: Make PipeAwaitable a non-nested internal struct instead of using partial classes.